### ISSUE TITLE
@GetIgnore doesn't work on companion object

### ISSUE THREAD

**Issue by mystdeim** [vert-x3/vertx-lang-kotlin#93]
After I upgraded to 3.6.0.CR1 getignore annotation was broken

```
@ProxyGen
@VertxGen
interface AccountService {

    fun create(account: Account, handler: Handler<AsyncResult<Long>>)
    operator fun get(id: Long, handler: Handler<AsyncResult<Account>>)

    @GenIgnore
    companion object {

        val ADDRESS = "account"

        fun createService(vertx: Vertx): AccountService {
            return AccountServiceImpl(vertx)
        }

        fun createProxy(vertx: Vertx, address: String): AccountService {
            return AccountServiceVertxEBProxy(vertx, address)
        }
    }
}
```

After I run `mvn package` I get: `Could not generate model for Companion: type mystdeim.vertx_examples.mvn_kotlin_proxy.service.AccountService.Companion is not legal for use for a constant type in code generation`

There's a simple project: [https://github.com/mystdeim/vertx_examples/tree/master/mvn_kotlin_proxy](https://github.com/mystdeim/vertx_examples/tree/master/mvn_kotlin_proxy)

**Comment by rgmz** [comment]
This is an unfortunate regression since the alternatives to `@GenIgnore` are quite ugly. 
(See [this thread](https://github.com/vert-x3/vertx-guide-for-java-devs/issues/28#issuecomment-411896310))

I'll see if I can pinpoint the cause.

**Comment by gmariotti** [comment]
Hi @mystdeim, I'm looking into this and I will come up with an example of how to use `service-proxy` with Kotlin. My personal opinion is that you shouldn't use `companion object` for this use case and prefer, instead, top-level functions and constants

**Comment by rgmz** [comment]
> My personal opinion is that you shouldn't use `companion object` for this use case and prefer, instead, top-level functions and constants

To my knowledge, companion objects + `@JvmStatic` annotations are the only way to have static factory methods that are honoured by codegen.

Other potential solutions (prior to Kotlion 1.3) were discussed here:
https://github.com/vert-x3/vertx-guide-for-java-devs/issues/28#issuecomment-411896310

**Comment by rgmz** [comment]
This will be fixed in 4.1.0, per #329.

**Comment by tsegismont** [comment]
Fixed by #329
